### PR TITLE
Cisco Email Threat Defense MS Sentinel connector 

### PR DESCRIPTION
**Pull request by sepintocisco**
Required items, please complete
   
   Change(s):
   - New solution for Cisco ETD connector for MS Sentinel

   Reason for Change(s):
   - New solution for Cisco ETD connector for MS Sentinel
   
   Version Updated:
   - no
   
   Testing Completed:
   - Yes. Manually deployed and tested.
 
   Checked that the validations are passing and have addressed any issues that are present:
   - No

### CODE REVIEW

**Comment by sepintocisco** [comment]
@microsoft-github-policy-service agree company="Cisco Systems"


From: microsoft-github-policy-service[bot] ***@***.***>
Date: Wednesday, 14 February 2024 at 17:48
To: Azure/Azure-Sentinel ***@***.***>
Cc: Sergio Pinto (sepinto) ***@***.***>, Mention ***@***.***>
Subject: Re: [Azure/Azure-Sentinel] Cisco Email Threat Defense MS Sentinel connector (PR #9957)

@sepintocisco<https://github.com/sepintocisco> please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.

@microsoft-github-policy-service agree [company="{your company}"]

Options:

  *   (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.

@microsoft-github-policy-service agree

  *   (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.

@microsoft-github-policy-service agree company="Microsoft"
Contributor License Agreement
Contribution License Agreement

This Contribution License Agreement (“Agreement”) is agreed to by the party signing below (“You”),
and conveys certain license rights to Microsoft Corporation and its affiliates (“Microsoft”) for Your
contributions to Microsoft open source projects. This Agreement is effective as of the latest signature
date below.

  1.  Definitions.
“Code” means the computer software code, whether in human-readable or machine-executable form,
that is delivered by You to Microsoft under this Agreement.
“Project” means any of the projects owned or managed by Microsoft and offered under a license
approved by the Open Source Initiative (www.opensource.org<http://www.opensource.org>).
“Submit” is the act of uploading, submitting, transmitting, or distributing code or other content to any
Project, including but not limited to communication on electronic mailing lists, source code control
systems, and issue tracking systems that are managed by, or on behalf of, the Project for the purpose of
discussing and improving that Project, but excluding communication that is conspicuously marked or
otherwise designated in writing by You as “Not a Submission.”
“Submission” means the Code and any other copyrightable material Submitted by You, including any
associated comments and documentation.
  2.  Your Submission. You must agree to the terms of this Agreement before making a Submission to any
Project. This Agreement covers any and all Submissions that You, now or in the future (except as
described in Section 4 below), Submit to any Project.
  3.  Originality of Work. You represent that each of Your Submissions is entirely Your original work.
Should You wish to Submit materials that are not Your original work, You may Submit them separately
to the Project if You (a) retain all copyright and license information that was in the materials as You
received them, (b) in the description accompanying Your Submission, include the phrase “Submission
containing materials of a third party:” followed by the names of the third party and any licenses or other
restrictions of which You are aware, and (c) follow any other instructions in the Project’s written
guidelines concerning Submissions.
  4.  Your Employer. References to “employer” in this Agreement include Your employer or anyone else
for whom You are acting in making Your Submission, e.g. as a contractor, vendor, or agent. If Your
Submission is made in the course of Your work for an employer or Your employer has intellectual
property rights in Your Submission by contract or applicable law, You must secure permission from Your
employer to make the Submission before signing this Agreement. In that case, the term “You” in this
Agreement will refer to You and the employer collectively. If You change employers in the future and
desire to Submit additional Submissions for the new employer, then You agree to sign a new Agreement
and secure permission from the new employer before Submitting those Submissions.
  5.  Licenses.

  *   Copyright License. You grant Microsoft, and those who receive the Submission directly or
indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license in the
Submission to reproduce, prepare derivative works of, publicly display, publicly perform, and distribute
the Submission and such derivative works, and to sublicense any or all of the foregoing rights to third
parties.
  *   Patent License. You grant Microsoft, and those who receive the Submission directly or
indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license under
Your patent claims that are necessarily infringed by the Submission or the combination of the
Submission with the Project to which it was Submitted to make, have made, use, offer to sell, sell and
import or otherwise dispose of the Submission alone or with the Project.
  *   Other Rights Reserved. Each party reserves all rights not expressly granted in this Agreement.
No additional licenses or rights whatsoever (including, without limitation, any implied licenses) are
granted by implication, exhaustion, estoppel or otherwise.

  1.  Representations and Warranties. You represent that You are legally entitled to grant the above
licenses. You represent that each of Your Submissions is entirely Your original work (except as You may
have disclosed under Section 3). You represent that You have secured permission from Your employer to
make the Submission in cases where Your Submission is made in the course of Your work for Your
employer or Your employer has intellectual property rights in Your Submission by contract or applicable
law. If You are signing this Agreement on behalf of Your employer, You represent and warrant that You
have the necessary authority to bind the listed employer to the obligations contained in this Agreement.
You are not expected to provide support for Your Submission, unless You choose to do so. UNLESS
REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING, AND EXCEPT FOR THE WARRANTIES
EXPRESSLY STATED IN SECTIONS 3, 4, AND 6, THE SUBMISSION PROVIDED UNDER THIS AGREEMENT IS
PROVIDED WITHOUT WARRANTY OF ANY KIND, INCLUDING, BUT NOT LIMITED TO, ANY WARRANTY OF
NONINFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
  2.  Notice to Microsoft. You agree to notify Microsoft in writing of any facts or circumstances of which
You later become aware that would make Your representations in this Agreement inaccurate in any
respect.
  3.  Information about Submissions. You agree that contributions to Projects and information about
contributions may be maintained indefinitely and disclosed publicly, including Your name and other
information that You submit with Your Submission.
  4.  Governing Law/Jurisdiction. This Agreement is governed by the laws of the State of Washington, and
the parties consent to exclusive jurisdiction and venue in the federal courts sitting in King County,
Washington, unless no federal subject matter jurisdiction exists, in which case the parties consent to
exclusive jurisdiction and venue in the Superior Court of King County, Washington. The parties waive all
defenses of lack of personal jurisdiction and forum non-conveniens.
  5.  Entire Agreement/Assignment. This Agreement is the entire agreement between the parties, and
supersedes any and all prior agreements, understandings or communications, written or oral, between
the parties relating to the subject matter hereof. This Agreement may be assigned by Microsoft.

**Comment by v-prasadboke** [comment]
Hello @sepintocisco, Thanks for raising this PR. This PR will be investigated and we will come back to you with an update by 21 February, 2024

File: Solutions/CiscoETD/Workbooks/CiscoETD.json
Comment (v-prasadboke) [inline Solutions/CiscoETD/Workbooks/CiscoETD.json:144]: Please change value of fromTemplateId. It shouldnt be sentinel-UserWorkbook

File: Solutions/CiscoETD/Data Connectors/CiscoETD_API_FunctionApp.json
Comment (v-prasadboke) [inline Solutions/CiscoETD/Data Connectors/CiscoETD_API_FunctionApp.json:1]: Please add custom field in Data connector 
you can refer to this connector 
Solutions/Agari/Data Connectors/Agari_API_FunctionApp.json

**Comment by v-prasadboke** [comment]
Hello @sepintocisco, Can you provide me write access to your branch. You can invite me as an collaborator. Thanks

**Comment by sepintocisco** [comment]
Done. Thanks for your support.

From: v-prasadboke ***@***.***>
Date: Tuesday, 20 February 2024 at 12:45
To: Azure/Azure-Sentinel ***@***.***>
Cc: Sergio Pinto (sepinto) ***@***.***>, Mention ***@***.***>
Subject: Re: [Azure/Azure-Sentinel] Cisco Email Threat Defense MS Sentinel connector (PR #9957)

Hello @sepintocisco<https://github.com/sepintocisco>, Can you provide me write access to your branch. You can invite me as an collaborator. Thanks

File: Solutions/CiscoETD/Data Connectors/CiscoETD_API_FunctionApp.json
Comment (v-prasadboke) [reply Solutions/CiscoETD/Data Connectors/CiscoETD_API_FunctionApp.json:1]: > Please add custom field in Data connector you can refer to this connector Solutions/Agari/Data Connectors/Agari_API_FunctionApp.json

Hello @sepintocisco, Please work on this requested commit

**Comment by v-prasadboke** [comment]
Please add workbook metadata in file Workbooks/WorkbooksMetadata.json

**Comment by v-prasadboke** [comment]
Hello @sepintocisco, can you try to resolve the new validation error for function app

**Comment by v-prasadboke** [comment]
Hello @sepintocisco, Please add workbook images with black background as well. 
Also please take a look at data connector validation. Permissions are not matching

File: Solutions/Cisco ETD/Data Connectors/azuredeploy_CiscoETD_API_FunctionApp.json
Comment (v-prasadboke) [inline Solutions/Cisco ETD/Data Connectors/azuredeploy_CiscoETD_API_FunctionApp.json:318]: Hello @sepintocisco, Link mentioned over here for zip file does not redirects to AZure repo.        
Can you explain me about the link mentioned by you

File: Solutions/Cisco ETD/Data Connectors/azuredeploy_CiscoETD_API_FunctionApp.json
Comment (v-prasadboke) [inline Solutions/Cisco ETD/Data Connectors/azuredeploy_CiscoETD_API_FunctionApp.json:6]: Hello @sepintocisco, Please concat function app name with resource group or subscription id. 
Reason being There can be multiple storage account with similar names. 

You can create a Variable, And put value as to concat functionapp name with other string. 

Please refer to this data connector for more clarification. 
Thanks.

File: Solutions/Cisco ETD/Data Connectors/azuredeploy_CiscoETD_API_FunctionApp.json
Comment (sahana-bannigol-cisco) [reply Solutions/Cisco ETD/Data Connectors/azuredeploy_CiscoETD_API_FunctionApp.json:318]: Hi, it is the same zip file present in the PR, that is hosted elsewhere. It needs to be changed to the github url that directs to the master branch of the repository, thus have kept it on hold. The same needs to be done for the url the deploy button redirects to. If you could guide us with the link to master branch, it would be of great help.

File: Solutions/Cisco ETD/Data Connectors/azuredeploy_CiscoETD_API_FunctionApp.json
Comment (sahana-bannigol-cisco) [reply Solutions/Cisco ETD/Data Connectors/azuredeploy_CiscoETD_API_FunctionApp.json:6]: Sure, will do it.

File: Solutions/Cisco ETD/Data Connectors/azuredeploy_CiscoETD_API_FunctionApp.json
Comment (sahana-bannigol-cisco) [reply Solutions/Cisco ETD/Data Connectors/azuredeploy_CiscoETD_API_FunctionApp.json]: Can the "WEBSITE_RUN_FROM_PACKAGE" url be from the repositories zip file? like this url "https://github.com/Cisco-Email-Security/Azure-Sentinel/blob/CiscoETD/Solutions/Cisco%20ETD/Data%20Connectors/CiscoETDAzureFunction.zip" but it should point to the Azure/Azure-Sentinel master branch

Comment (v-prasadboke) [reply Solutions/Cisco ETD/Data Connectors/azuredeploy_CiscoETD_API_FunctionApp.json]: Hello @sepintocisco, We use shortlink (aka.ms) which redirects to the actual hyperlink of the repo. I'll update the link later, Dont worry

File: Solutions/Cisco ETD/SolutionMetadata.json
Comment (v-prasadboke) [inline Solutions/Cisco ETD/SolutionMetadata.json:2]: Please confirm if the solution is to be published by Microsoft. 
Haven't received any quote on publishing this Offer from our side.

If it is to be published from your end, Then change the support info

Comment (v-prasadboke) [reply Solutions/Cisco ETD/SolutionMetadata.json:2]: Please change publisher id as well. It should be the one which your Partner center supports.

Comment (sahana-bannigol-cisco) [reply Solutions/Cisco ETD/SolutionMetadata.json]: The changes are made.

**Comment by v-prasadboke** [comment]
Hi @sepintocisco, everything looks good. 
Just I'm missing with credentials to test the content of the solution. If possible can you share the working/result images of function app. 
Just this and we are good to merge the PR

**Comment by sepintocisco** [comment]
@v-prasadboke , how can I share that information with you?

**Comment by v-prasadboke** [comment]
You share the working images over here in comments

**Comment by sahana-bannigol-cisco** [comment]
> You share the working images over here in comments

Please find the attached images
ConnectorPage
FunctionApp
LogAnalytics

**Review by v-prasadboke — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**
